allennlp_rc/dataset_readers/transformer_squad.py

```diff
@@ -0,0 +1,323 @@
+        implementation.
+    skip_invalid_examples: ``bool``, optional (default=False)
+        If this is true, we will skip examples that don't have a gold answer. You should set this to True during
+        training, and False any other time.
+    max_query_length : ``int``, optional (default=64)
+        The maximum number of wordpieces dedicated to the question. If the question is longer than this, it will be
+        truncated.
+    max_questions : ``int``, optional (default=None)
+        If this is specified, the reader quits after this many questions have been processed. This can be useful
+        for debugging.
+    """
+
+    def __init__(
+        self,
+        transformer_model_name: str = "bert-base-cased",
+        length_limit: int = 384,
+        stride: int = 128,
+        skip_invalid_examples: bool = False,
+        max_query_length: int = 64,
+        max_questions: Optional[int] = None,
+    ) -> None:
+        super().__init__()
+        self._tokenizer = PretrainedTransformerTokenizer(
+            transformer_model_name, add_special_tokens=False, calculate_character_offsets=True
+        )
+        self._token_indexers = {"tokens": PretrainedTransformerIndexer(transformer_model_name)}
+        self.length_limit = length_limit
+        self.stride = stride
+        self.skip_invalid_examples = skip_invalid_examples
+        self.max_query_length = max_query_length
+        self.max_questions = max_questions
+        if (
+            AutoModel.from_pretrained(
+                transformer_model_name
+            ).embeddings.token_type_embeddings.num_embeddings
+            > 1
+        ):
+            self.non_content_type_id = 1
+        else:
+            self.non_content_type_id = 0
+
+    @overrides
+    def _read(self, file_path: str):
+        # if `file_path` is a URL, redirect to the cache
+        file_path = cached_path(file_path)
+
+        logger.info("Reading file at %s", file_path)
+        open_fn: Callable[[str], IO[Any]] = open
+        if file_path.endswith(".gz"):
+            import gzip
+
+            open_fn = gzip.open
+        elif file_path.endswith(".bz2"):
+            import bz2
+
+            open_fn = bz2.open
+        with open_fn(file_path) as dataset_file:
+            dataset_json = json.load(dataset_file)
+            dataset = dataset_json["data"]
+        logger.info("Reading the dataset")
+        question_count = 0
+        questions_with_more_than_one_instance = 0
+        for article in dataset:
+            for paragraph_json in article["paragraphs"]:
+                context = paragraph_json["context"]
+                for question_answer in paragraph_json["qas"]:
+                    answers = [answer_json["text"] for answer_json in question_answer["answers"]]
+
+                    # This version does not support no-answer questions.
+                    if len(answers) <= 0:
+                        continue
```

Comment:
Maybe log once or maintain a count of these and log it at the end of `read`?

Reply:
ada18a20b54e14c9a5941706e74fd4fc7c615212